src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -217,37 +251,87 @@ private void ShowGoldBar()
                               action: InfoBarClose));
         }
 
-        private ReSharperStatus IsReSharperEnabled()
+        /// <summary>
+        /// Returns true if ReSharper is installed, enabled, and not suspended.  
+        /// </summary>
+        private async ValueTask<ReSharperStatus> IsReSharperRunningAsync(ReSharperStatus lastStatus, CancellationToken cancellationToken)
         {
-            AssertIsForeground();
-
             // Quick exit if resharper is either uninstalled or not enabled
-            if (!_resharperExtensionEnabled)
+            if (!_resharperExtensionInstalledAndEnabled)
             {
                 return ReSharperStatus.NotInstalledOrDisabled;
             }
 
-            if (_oleCommandTarget == null)
+            await EnsureOleCommandTargetAsync().ConfigureAwait(false);
+
+           // poll until either suspend or resume botton is available, or until operation is canceled
+            while (true)
             {
-                _oleCommandTarget = _serviceProvider.GetService<IOleCommandTarget, SUIHostCommandDispatcher>();
-            }
+                cancellationToken.ThrowIfCancellationRequested();
 
-            var cmds = new OLECMD[1];
-            cmds[0].cmdID = SuspendId;
-            cmds[0].cmdf = 0;
+                var suspendFlag = await QueryStatusAsync(SuspendId).ConfigureAwait(false);
 
-            var hr = _oleCommandTarget.QueryStatus(ReSharperCommandGroup, (uint)cmds.Length, cmds, IntPtr.Zero);
-            if (ErrorHandler.Failed(hr))
-            {
                 // In the case of an error when attempting to get the status, pretend that ReSharper isn't enabled. We also
                 // shut down monitoring so we don't keep hitting this.
-                FatalError.ReportWithoutCrash(Marshal.GetExceptionForHR(hr));
-                Shutdown();
-                return ReSharperStatus.NotInstalledOrDisabled;
+                if (suspendFlag == 0)
+                {
+                    return ReSharperStatus.NotInstalledOrDisabled;
+                }
+
+                var resumeFlag = await QueryStatusAsync(ResumeId).ConfigureAwait(false);
+                if (resumeFlag == 0)
+                {
+                    return ReSharperStatus.NotInstalledOrDisabled;
+                }
+
+                // When ReSharper is suspended, the ReSharper_Resume command is Enabled and not Invisible
```

Comment:
Consider moving this up to before the resume flag query, so we don't query that if we don't have to.